src/field_theory/separable_degree.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Scholbach
+-/
+import field_theory.separable
+import algebra.algebra.basic
+import data.polynomial.degree
+import algebra.char_p.exp_char
+/-!
+# Separable degree
+
+This file contains basics about the separable degree of a polynomial.
+
+## Main results
+- `separable_degree`: the definition of the separable degree
+- `irreducible_has_separable_contraction`: any nonzero irreducible polynomial can be contracted
+  to a separable polynomial
+- `separable_degree_dvd_degree'`: the separable degree divides the degree, in function of the
+  exponential characteristic of the field
+- `separable_degree_dvd_degree` and `separable_degree_eq_degree` specialize the statement of
+  `separable_degree_dvd_degree`
+
+## Tags
+separable degree, degree, polynomial
+-/
+
+noncomputable theory
+open_locale classical
+
+variables {F : Type} [comm_semiring F]
+  (q : ℕ) [exp_char F q]
+
+/-- A separable contraction of a polynomial `f` is a separable polynomial `g` such that
+`g(x^(q^m)) = f(x)` for some `m : ℕ`.-/
+class has_separable_contraction (f : polynomial F) : Prop :=
```

Comment:
I think this *can* be a class, but it's probably better to just make it a regular definition.

Reply:
Done.